### ISSUE TITLE
skip the "pr" build if the pull request is a fast-forward

### ISSUE THREAD

**Issue by afeld** [travis-ci/travis-ci#3340]
While I appreciate the complimentary extra build to ensure that the code will work once merged, there is effectively no difference between the "pr" build and the "push" one if the merge will (effectively) be a fast-forward. This would reduce load on Travis (no need for redundant builds), and also reduce turn-around time waiting for the commit status to be green. Thanks!

![screen shot 2015-03-06 at 11 55 40 am](https://cloud.githubusercontent.com/assets/86842/6529711/99c61510-c3f8-11e4-938d-b4a2ab24f330.png)

**Comment by rkh** [comment]
Would make a lot of sense. Not sure we can easily tell it's ff, I think GitHub always generates a merge commit.

**Comment by afeld** [comment]
Right, merging through the web interface creates a merge commit... I just mean if it *could* be a FF merge, i.e. that there are no newer commits on the destination branch.

**Comment by rkh** [comment]
Yeah, we'd have to check the git history via the GitHub API, if we could get that down to a very small number of requests (one to three or so) that should be doable.

**Comment by rkh** [comment]
The check from the PR event if we have a push even might also be a bit more complex but should be possible.

**Comment by thedrow** [comment]
Ref #1504 which was closed due to the complexity of this feature.
I'd be happy to see it implemented though.

**Comment by ypresto** [comment]
This is really problem when git-flow like branching policy is used.

We have PRs like below:

```
develop <- feature(s)
master <- develop
master <- hotfix
```

We want build commits of master branch and all PRs (including PRs for develop branch).
Or want to skip build on `master <- develop` PR (use commit CI instead).

But there is no way to specify "commit only branch" and "PR only branch".
`branches: only:` causes both push and PR build to be fired.

**Comment by joshk** [comment]
Thanks for the extra info, I will bring this to the team so we can discuss options.

I don't have an ETA sorry.

**Comment by cancan101** [comment]
I understand that there may be cases when the scripts key off of the PR ENV variable but it be nice to have the option to suppress PR builds when there is an FF merge.

**Comment by cskiwi** [comment]
No updates on this? it's been almost 3 years ...

**Comment by edmorley** [comment]
This is still relevant and would help reduce load by a reasonable amount.

**Comment by edmorley** [comment]
> This is still relevant and would help reduce load by a reasonable amount.
